rclcpp/include/rclcpp/intra_process_manager_impl.hpp

```diff
@@ -283,7 +283,7 @@ class IntraProcessManagerImpl : public IntraProcessManagerImplBase
   fixed_size_string(const char * str) const
   {
     FixedSizeString ret;
-    std::strncpy(ret.data(), str, ret.size());
```

Comment:
The previous code already looks wrong. I always copies `ret.size()` bytes - even if `str` is shorter much implies copying undefined data.

This should check the `strlen(str)`, ensure that it is <= `RMW_TOPIC_MAX_NAME_LENGTH` and only copy its actual data.

Reply:
Actually, std::strncpy pads with zeros after the '\0'. I'm using now rcutils_snprintf, which stops copying in the null terminator and solves the same portability issue.
I'm adding the strcmp_wrapper again, as the lexicographical comparison of std::array doesn't stop in the null terminator. I didn't realize before, and as std::strncpy pads with zeros, it was passing.

Reply:
Why use the complexity of `snprintf` if all needed is a 1-to-1 copy?